tests/test_injection.py

```diff
@@ -34,3 +35,11 @@ async def inner(
 
     with pytest.raises(RuntimeError, match="Injected arguments must not be redefined"):
         await inner(_=container.SimpleFactory(dep1="1", dep2=2))
+
+
+def test_type_check() -> None:
+    @inject
+    async def main() -> None:
+        pass
+
+    asyncio.run(main())
```

Comment:
This is just a test to show that `mypy` complains about the following before the proposed changes in `inject`:
```
tests/test_injection.py:45: error: Argument 1 to "run" has incompatible type "Awaitable[None]"; expected "Coroutine[Any, Any, Never]"  [arg-type]
Found 1 error in 1 file (checked 22 source files)
task: Failed to run task "lint": exit status 1
```